worksheet.rb

```diff
@@ -1,34 +1,144 @@
-# Which layers are nested in each other?
-# Which layers of data "have" within it a different layer?
-# Which layers are "next" to each other?
+#   Drivers, dates, fares, riders, and ratings
+#
+# The outermost layer is the entire table, the list of rides.
+#   Inside, 'next' to each other, will be the driver hashes. With driver and ride information.
+#   Inside the ride information will be another layer, with a list of rides.
+#   Inside the rides will be the ride details, with the rider, fare, date, and rating information running parallel.
 
 ########################################################
 # Step 2: Assign a data structure to each layer
-
-# Copy your list from above, and in this section
-# determine what data structure each layer should have
+#
+# The outermost layer is the entire table, the list of rides. Rides = {}
+#   Inside, 'next' to each other, will be the driver hashes. With driver and ride information. :DriverID, ride_info
+#     Inside the ride information will be another layer, with a list of rides. ride_info = [multiple rides]
+#       Inside the rides will be the ride details, with the rider, fare, date, and rating information running parallel. ride = {:RiderID, :$, :*}
 
 ########################################################
 # Step 3: Make the data structure!
 
-# Setup the entire data structure:
-# based off of the notes you have above, create the
-# and manually write in data presented in rides.csv
-# You should be copying and pasting the literal data
-# into this data structure, such as "DR0004"
-# and "3rd Feb 2016" and "RD0022"
+Rideshare = {
+    DR0004: [
+        {Date: "3rd Feb 2016",
+         Fare: 5,
+         Rider: "RD0022",
+         Rating: 5
+        },
+        {Date: "4th Feb 2016",
+         Fare: 10,
+         Rider: "RD0022",
+         Rating: 4
+        },
+        {Date: "5th Feb 2016",
+         Fare: 20,
+         Rider: "RD0073",
+         Rating: 5
+        }
+    ],
+    DR0003: [
+    {Date: "4th Feb 2016",
+     Fare: 5,
+     Rider: "RD0066",
+     Rating: 5
+    },
+    {Date: "5th Feb 2016",
+     Fare: 50,
+     Rider: "RD0003",
+     Rating: 2
+    }
+],
+    DR0002: [
+    {Date: "3rd Feb 2016",
+     Fare: 25,
+     Rider: "RD0073",
+     Rating: 5
+    },
+    {Date: "4th Feb 2016",
+     Fare: 15,
+     Rider: "RD0013",
+     Rating: 1
+    },
+    {Date: "5th Feb 2016",
+     Fare: 35,
+     Rider: "RD0066",
+     Rating: 3
+    }
+],
+    DR0001: [
+    {Date: "3rd Feb 2016",
+     Fare: 10,
+     Rider: "RD0003",
+     Rating: 3
+    },
+    {Date: "3rd Feb 2016",
+     Fare: 30,
+     Rider: "RD0015",
+     Rating: 4
+    },
+    {Date: "5th Feb 2016",
+     Fare: 45,
+     Rider: "RD0003",
+     Rating: 2}
+]
+}
 
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
 
 # Use an iteration blocks to print the following answers:
+
 # - the number of rides each driver has given
+def number_of_rides (driver)
```

Comment:
Style: in the future avoid having a space between the method name and the parameters.

```suggestion
def number_of_rides(driver)
```